tests/manage/test_reclaim_policy.py

```diff
@@ -0,0 +1,94 @@
+import logging
+import pytest
+from ocs_ci.framework.testlib import tier1, ManageTest
+from ocs_ci.ocs.resources import pod
+from tests import helpers
+from ocs_ci.ocs import ocp
+
+
+log = logging.getLogger(__name__)
+
+
+@pytest.fixture(scope='class')
+def test_fixture(request):
+    """
+    This is a test fixture
+    """
+    self = request.node.cls
+
+    def finalizer():
+        teardown(self)
+
+    request.addfinalizer(finalizer)
+    setup(self)
+
+
+def setup(self):
+    """
+    Setting up a secret and storage class
+    """
+    self.secret_obj = helpers.create_secret('CephBlockPool')
+    self.sc_obj_retain = helpers.create_storage_class(
+        interface_type='CephBlockPool', interface_name='rbd',
+        secret_name=self.secret_obj.name, reclaim_policy='Retain'
+    )
+    self.sc_obj_delete = helpers.create_storage_class(
+        interface_type='CephBlockPool', interface_name='rbd',
+        secret_name=self.secret_obj.name, reclaim_policy='Delete'
+    )
+
+
+def teardown(self):
+    """
+    Deleting secret
+    """
+    assert self.sc_obj_retain.delete()
+    assert self.sc_obj_delete.delete()
+    assert self.secret_obj.delete()
+
+
+def list_ceph_images(pool_name='rbd'):
+    """
+    Args:
+        pool_name: (str) Name of the pool to get the ceph images
+
+    Returns: (List) of RBD images in the pool
+    """
+    ct_pod = pod.get_ceph_tools_pod()
+    return ct_pod.exec_ceph_cmd(ceph_cmd=f"rbd ls {pool_name}", format='json')
+
+
+@tier1
+@pytest.mark.usefixtures(test_fixture.__name__)
+class TestReclaimPolicy(ManageTest):
+
+    def test_reclaim_policy_retain(self):
+        """
+        Calling functions for pvc invalid name and size
+        """
+        log.info("sc_retain")
+        sc_name = self.sc_obj_retain.name
+        pvc_count = len(list_ceph_images())
+        pvc_obj = helpers.create_pvc(sc_name, pvc_name="pvc-retain")
+        pv_name = pvc_obj.get()['spec']['volumeName']
```

Comment:
pv_name = pvc_obj.get('spec').get('volumeName')

Reply:
This doesn't seem to work.  pvc_obj.get('spec') still returns what pvc_obj.get() returns. 

(Pdb) pvc_obj.get('spec')
19:26:52 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/kramdoss/ocs-ci-new12/auth/kubeconfig get PersistentVolumeClaim pvc-retain -o yaml
{'apiVersion': 'v1', 'kind': 'PersistentVolumeClaim', 'metadata': {'annotations': {'pv.kubernetes.io/bind-completed': 'yes', 'pv.kubernetes.io/bound-by-controller': 'yes', 'volume.beta.kubernetes.io/storage-provisioner': 'rbd.csi.ceph.com'}, 'creationTimestamp': '2019-07-04T13:54:12Z', 'finalizers': ['kubernetes.io/pvc-protection'], 'name': 'pvc-retain', 'namespace': 'openshift-storage', 'resourceVersion': '1660121', 'selfLink': '/api/v1/namespaces/openshift-storage/persistentvolumeclaims/pvc-retain', 'uid': '33f842ea-9e63-11e9-9c94-06621664a1c4'}, 'spec': {'accessModes': ['ReadWriteOnce'], 'dataSource': None, 'resources': {'requests': {'storage': '3Gi'}}, 'storageClassName': 'storageclass-test-0419232104', 'volumeMode': 'Filesystem', 'volumeName': 'pvc-33f842ea-9e63-11e9-9c94-06621664a1c4'}, 'status': {'accessModes': ['ReadWriteOnce'], 'capacity': {'storage': '3Gi'}, 'phase': 'Bound'}}